cpp/src/io/orc/reader_impl.cu

```diff
@@ -933,8 +933,8 @@ std::unique_ptr<table> reader::impl::compute_timezone_table(
     {}, selected_stripes[0].stripe_info[0].second->writerTimezone, stream);
 }
 
-table_with_metadata reader::impl::read(size_type skip_rows,
-                                       size_type num_rows,
+table_with_metadata reader::impl::read(int64_t skip_rows,
```

Comment:
Worth creating a `cudf::io::row_index_type` (please suggest a better name) alias instead of manually switching to `int64_t` all over the place.

Reply:
Well. I tried to do this and ran into an issue with our file/include structure. This type would be needed on both `/include` and `/src` side, but there are no header files that are included in both.
I think we need to create a new header that only contains ORC "specs" and declare the type(s) there. I think this can be done in the same PR as https://github.com/rapidsai/cudf/pull/13063#discussion_r1160948892

Reply:
Works for me. Thanks for looking into this.